product-service/handlers/getProductById.ts

```diff
@@ -0,0 +1,35 @@
+import fetch from "node-fetch";
+import {APIGatewayProxyHandler} from "aws-lambda";
+import products from "../products";
+
+const headers = {
+    'Access-Control-Allow-Origin': '*',
+}
+
+export const getProductById: APIGatewayProxyHandler = async (event, _context) => {
+    const {productId} = event.pathParameters;
```

Comment:
This line may result in a failure.

- no try/catch around the business logic
- `event.pathParameters` can be null:
```
export interface APIGatewayProxyEventBase<TAuthorizerContext> {
    ...
    pathParameters: { [name: string]: string } | null;
    ...
}
```

Reply:
@bhondu thanks! fixed it